python/shotgun_fields/bubble_widget.py

```diff
@@ -0,0 +1,495 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+from sgtk.platform.qt import QtCore, QtGui
+
+
+class BubbleWidget(QtGui.QFrame):
+    """
+    This class represents a drawable "bubble" ot display in a :class:`.BubbleEditWidget`
+
+    This widget will stores data for the object it represents. The data can be set
+    and accessed via the respective ``get_data()`` and ``set_data()`` methods.
+
+    The widget can display an optional image along with its text. See the display
+    related methods ``set_text()`` and ``set_image()`` below.
+
+    :signal: ``remove_clicked()`` - emitted when the widget's ``x`` button clicked.
+    """
+
+    # signal emitted when the widget's remove button was clicked
+    remove_clicked = QtCore.Signal()
+
+    def __init__(self, parent=None):
+        """Initialize the widget.
+
+        :param parent: This widget's parent widget
+        :type parent: :class:`~PySide.QtGui.QWidget`
+        """
+        super(BubbleWidget, self).__init__(parent)
+
+        # placeholder for the underlying data this widget represents in the editor
+        self._data = None
+
+        # should not grow or shrink
+        self.setSizePolicy(QtGui.QSizePolicy.Fixed, QtGui.QSizePolicy.Fixed)
+
+        # style the look of the bubble
+        self.setObjectName("bubble")
```

Comment:
I'll defer to @robblau as I think he added this bit. perhaps he tried setting stylesheet without limiting it to a named widget? i will see if it works without naming, but don't think it will hurt to leave as is... will run some tests.